autopush/base.py

```diff
@@ -45,3 +45,30 @@ def write_error(self, code, **kwargs):
             self.log.failure("Error in handler: %s" % code,
                              client_info=self._client_info)
         self.finish()
+
+    def authenticate_peer_cert(self):
+        """Authenticate the client per the configured peer certs.
+
+        Aborts the request w/ a 401 on failure.
+
+        """
+        cert = self.request.connection.transport.getPeerCertificate()
+        if cert:
+            sha256 = cert.digest('sha256')
```

Comment:
nit: can we call this something more explicit? (e.g. `cert_signature`?) Just think there's a small risk of collision with the hash function.